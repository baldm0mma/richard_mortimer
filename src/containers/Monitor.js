import React from 'react';
import { connect } from 'react-redux';
import Typist from 'react-typist';
import { Link } from 'react-router-dom';
import './Monitor.css';

export const Monitor = ({ userName }) => {
  const login = () => {
    return (
      <section className='type_container'>
        <Typist avgTypingDelay={30}>
          {`Hello Officer ${userName}, your Interdimentional Criminal Code Enforcement Organization login credentials have been accepted...`}
          <br />
          <Typist.Delay ms={1000} />
          {`Please wait while we log you in...`}
          <br />
          <Typist.Delay ms={1000} />
          {`...`}
          <br />
          <Typist.Delay ms={1000} />
          {`...`}
          <br />
          <Typist.Delay ms={1000} />
          {`...`}
          <br />
          <Typist.Delay ms={1500} />
          {`Login successful`}
          <br />
          <br />
          <Typist.Delay ms={1000} />
          {`I am G.L.E.A, the AI in control of this GALACTIC LAW ENFORCEMENT ATLAS terminal.`}
          <br />
          <Typist.Delay ms={1000} />
          {`My database contains information on some of the most brutalizing and vile criminals in the known multiverse.`}
          <br />
          <br />
          <Typist.Delay ms={2000} />
          {`${userName}, your active status has been reinstated by the Intergalactic Council on Multiverse Security and Defense to assist in the SWIFT DISTRIBUTION OF JUSTICE AND PUNISHMENT to these notorious villains.`}
          <br />
          <br />
          <Typist.Delay ms={1000} />
          <Link to='/criminals' className='proceed-link'>
            Proceed to database?
          </Link>
        </Typist>
      </section>
    );
  };

  return <section>{userName && login()}</section>;
};

export const mapStateToProps = ({ userName, page }) => ({
  userName,
  page
});

export default connect(
  mapStateToProps,
  null
)(Monitor);
