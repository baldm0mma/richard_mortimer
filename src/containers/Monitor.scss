@keyframes typing {
  from {
    width: 100%;
  }
  to {
    width: 0;
  }
}

.container {
  position: relative;
  width: 60%;
  margin: 0 auto;
  margin-top: 200px;
  font-family: Consolas, monospace;
}

.typing {
  position: absolute;
  top: 0;
  margin: 0;
  z-index: -1;
}

.hiders {
  margin: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

.hiders p {
  position: relative;
  clear: both;
  margin: 0;
  float: right; /* makes animation go left-to-right */
  width: 0; /* graceful degradation: if animation doesn't work, these are invisible by default */
  background: white; /* same as page background */
  animation: typing 4s steps(100, end);
  animation-fill-mode: both; /* load first keyframe on page load, leave on last frame at end */
}

.hiders p:nth-child(2) {
  animation-delay: 4s;
}

.hiders p:nth-child(3) {
  animation-delay: 8s;
}

.hiders p:nth-child(4) {
  animation-delay: 12s;
}

.hiders p:nth-child(5) {
  animation-delay: 16s;
}
