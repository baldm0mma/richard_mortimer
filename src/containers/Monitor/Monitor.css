.type_container {
  width: 80%;
  margin: 0 auto;
  padding-top: 75px;
  text-align: center;
  font-family: 'VT323', monospace;
  font-size: 2rem;
  color: #7edf7e;
  text-shadow: #7edf7e 1px 0 10px;
}

.proceed-link {
  animation: blinker 1s linear infinite;
  text-decoration: none;
  color: #7edf7e;
}

@keyframes blinker {
  50% {
    opacity: 0;
  }
}