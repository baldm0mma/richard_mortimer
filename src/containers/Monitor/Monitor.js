import './Monitor.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import Typist from 'react-typist';

export class Monitor extends Component {
  login = () => {
    return (
      <>
        <header className='type_header'>
          <Link to='/criminals'>
            <p className='skip_intro'>Skip Intro =></p>
          </Link>
        </header>
        <section className='type_container'>
          <Typist avgTypingDelay={40}>
            {`Hello Officer ${
              this.props.userName
            }, your Interdimentional Criminal Code Enforcement Organization login credentials have been accepted...`}
            <br />
            <Typist.Delay ms={1000} />
            {`Please wait while we log you in...`}
            <br />
            <Typist.Delay ms={1000} />
            {`...`}
            <br />
            <Typist.Delay ms={1000} />
            {`...`}
            <br />
            <Typist.Delay ms={1000} />
            {`...`}
            <br />
            <Typist.Delay ms={1500} />
            {`Login successful`}
            <br />
            <br />
            <Typist.Delay ms={1000} />
            {`I am G.L.E.A, the AI in control of this GALACTIC LAW ENFORCEMENT ATLAS terminal.`}
            <br />
            <Typist.Delay ms={1000} />
            {`My database contains information on some of the most brutalizing and vile criminals in the known multiverse.`}
            <br />
            <br />
            <Typist.Delay ms={2000} />
            {`${
              this.props.userName
            }, your active status has been reinstated by the Intergalactic Council on Multiverse Security and Defense to assist in the SWIFT DISTRIBUTION OF JUSTICE AND PUNISHMENT to these notorious villains.`}
            <br />
            <br />
            <Typist.Delay ms={1000} />
            <Link to='/criminals' className='proceed-link'>
              Proceed to database?
            </Link>
          </Typist>
        </section>
      </>
    );
  };

  render = () => {
    return <section>{this.props.userName && this.login()}</section>;
  };
}

export const mapStateToProps = ({ userName }) => ({
  userName
});

export default connect(
  mapStateToProps,
  null
)(Monitor);
