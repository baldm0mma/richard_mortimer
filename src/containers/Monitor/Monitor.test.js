import React from 'react';
import { Monitor, mapStateToProps } from './Monitor';
import { shallow } from 'enzyme';

describe('Monitor', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Monitor />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should revieve props of the user name', () => {
    const mockState = {
      userName: 'Biff'
    };
    const expected = {
      userName: 'Biff'
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
