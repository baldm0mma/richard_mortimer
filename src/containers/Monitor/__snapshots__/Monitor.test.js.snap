// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitor should match the snapshot 1`] = `ShallowWrapper {}`;
